docker-compose.yml

```diff
@@ -0,0 +1,146 @@
+# Docker Compose configuration for Survey Dashboard
+# Orchestrates nginx reverse proxy and Panel application containers
+
+version: '3.8'
+
+services:
+  # Panel application container
+  dashboard:
+    build: .
+    # Builds from Dockerfile in current directory
+    # Uses multi-stage build to create optimized production image
+
+    expose:
+      - "5006"
+    # Exposes port 5006 to other containers (not to host machine)
+    # Only accessible within Docker network - nginx can reach it
+
+    restart: unless-stopped
+    # Automatically restart if container crashes
+    # Persists across system reboots unless manually stopped
+
+    environment:
+      - PYTHONUNBUFFERED=1
+      # Ensures Python output is sent directly to logs (not buffered)
+      # Makes 'docker-compose logs' show real-time output
+      - VIRTUAL_HOST=${HOST}
+      # Tells nginx-proxy which domain should route to this container
+      # nginx-proxy watches Docker and auto-configures when it sees this variable
+      - VIRTUAL_PATH=${VIRTUAL_PATH}
+      # Restricts access to only this specific path on the domain
+      # Example: VIRTUAL_PATH=/survey-dashboard means only /survey-dashboard/* is accessible
+      # All other paths on this domain return 404
+      - VIRTUAL_PROTO=http
+      # Protocol nginx-proxy uses to communicate with this container
+      # Uses HTTP inside Docker network (fast, safe), HTTPS for external traffic
+      - LETSENCRYPT_HOST=${HOST}
+      # Tells acme-companion to get SSL certificate for this domain
+      # acme-companion watches Docker and auto-requests certificates when it sees this
+      - LETSENCRYPT_EMAIL=${LETSENCRYPT_EMAIL}
+      # Email for Let's Encrypt certificate notifications
+      # Used for expiry warnings and security notices about your certificates
+
+    # Uncomment to mount local code for development
+    # volumes:
+    #   - ./survey_dashboard:/app/survey_dashboard
+
+  # Nginx reverse proxy container with auto-configuration
+  nginx-proxy:
```

Comment:
for development purposes, just an nginx server is enough here. No need for ngninx proxy and letsencrypt